instrumentation/micrometer/micrometer-1.5/library/src/main/java/io/opentelemetry/instrumentation/micrometer/v1_5/AsyncInstrumentRegistry.java

```diff
@@ -24,7 +24,10 @@
 // TODO: refactor this class, there's too much copy-paste here
 final class AsyncInstrumentRegistry {
 
-  private final Meter meter;
+  // using a weak ref so that the AsyncInstrumentRegistry (which is stored in a static maps) does
+  // not hold strong references to Meter (and thus make it impossible to collect Meter garbage).
+  // in practice this should never return null
```

Comment:
it's not obvious to me why `Meter` can't be GC'd before `AsyncInstrumentRegistry`, can you explain briefly, or what do you think about using weak value in the cache instead `Cache<Meter, WeakReference<AsyncInstrumentRegistry>>`?

Reply:
It can't be GC'd because the `OpenTelemetryMeterRegistry` maintains a strong reference to both `Meter` and `AsyncInstrumentRegistry` -- so if the registry instance is collected then the registry cannot possibly be used anymore. I'll add a comment.

> or what do you think about using weak value in the cache instead `Cache<Meter, WeakReference<AsyncInstrumentRegistry>>`?

Hmm, in case the meter registry is GC'd the `WeakReference<AsyncInstrumentRegistry>` could get nulled out even if the `Meter` is not GC'd yet (because the `OpenTelemetry` instance is still used) - this hypothetical scenario would break the async instruments, cause if you create a new `OpenTelemetryMeterRegistry` for the same `OpenTelemetry`/`Meter` the async instrument registry will get cleared out.

Reply:
oh, got it 👍

> I'll add a comment

❤️